packages/runtime/src/runtimes/react/components/LiveProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import { ReactNode, useEffect, useMemo } from 'react'
+
+import { StoreContext, storeContextDefaultValue } from '..'
+import * as ReactPage from '../../../state/react-page'
+import { MakeswiftProvider, MakeswiftClient } from '../../../api/react'
+import { registerDocumentEffect } from '../../../state/actions'
+
+type Props = {
+  client: MakeswiftClient
+  rootElements?: Map<string, ReactPage.Element>
+  children?: ReactNode
+}
+
+export default function LiveProvider({ client, children, rootElements }: Props): JSX.Element {
+  const store = useMemo(
+    () =>
+      ReactPage.configureStore({
+        preloadedState: storeContextDefaultValue.getState(),
+        rootElements,
+      }),
+    [],
```

Comment:
This is missing the `rootElements` dependency

Reply:
Added